vulkan_generator/vulkan_utils/parsing_utils.py

```diff
@@ -14,14 +14,45 @@
 
 """This module contains the utility functions that needed elsewhere while parsing Vulkan XML"""
 
+import os
 import xml.etree.ElementTree as ET
 
 
-def get_text_from_tag(root: ET.Element, tag: str) -> str:
+def get_text_from_tag(elem: ET.Element, tag: str) -> str:
     """Gets the text of the element with the given tag"""
-    for child in root:
+    value = try_get_text_from_tag(elem, tag)
+
+    if value is None:
+        # This should not happen
+        raise SyntaxError(f"No {tag} tag found in {ET.tostring(elem, 'utf-8')}")
+
+    return value
+
+def try_get_text_from_tag(elem: ET.Element, tag: str) -> str:
```

Comment:
> So if we are returning either a string or None from a function, then the function returns Optional[str]

Yes

> I thought that Optional would be a type, similar to java.

Not really. It is not a class that you can instantiate. It is just used for type hints. `mypy` is pretty clever with Optional:

```python
# x has type Optional[str]

if not x:
  raise ValueError()

# Here, mypy knows x has type `str` (i.e. x definitely is not None)

# Assume `def use_stuff(x: str) -> None`.
# If we did not do the `if` check above,
# then `mypy` would disallow the call below because
# we are passing `Optional[str]` but the function expects `str`.
use_stuff(x)

```

Reply:
👍